doc/macsec/MACsec_hld.md

```diff
@@ -0,0 +1,743 @@
+
+In all scenarios, both virtual and real switch, the IP address will be assigned to the Port. The MKA protocol traffics, EAPOL packets, sent by wpa_supplicant directly use the Port as the egress/ingress port but the SecY will be set as the bypass mode to the MKA protocol traffic, which means the SecY will not encrypt, decrypt or validate the EAPOL packets and directly deliver them to next interface. All traffics, except EAPOL packets, transmitted on the Port will be encrypted by SecY and then these traffics will be put to the physical port for transmission. While all traffics, except EAPOL packets, received on the physical port will be validated and decrypted and then these traffics will be delivered to the Port or discarded if the validation fails.
+
+![interface stack](images/interface_stack.png)  
+
+## 3 Modules Design
+
+The following figure depicts the data flow and related components of MACsec. Different colored boxes indicate the components that belong to different containers or spaces. The following paragraph only lists the necessary components that are needed by the functionality of MACsec.
+
+- The orange means these components belong to MACsec container. This container include the control plane of MACsec that include MACsec enable/disable, MACsec session negotiation and key exchange. And it consists of daemons, MACsecMgr and wpa_supplicant, and a tool, wpa_cli. Meanwhile MACsec container should be running after SWSS container start, because MACsec feature depends on the Orchagent that interacts with MACsec SAI for the management of SecY.
+  - **MACsecMgr** controls the availability of MACsec on specified interfaces by using wpa_cli commands to communicate with wpa_supplicant daemon.
+  - **wpa_supplicant** uses the MKA protocol to manage the MACsec session and key exchange in MACsec control plane. It calls the predefined MACsec plugin APIs to communicate with SONiC MACsec plugin.
+  - **SONiC MACsec Plugin** is a plugin of wpa_supplicant that follows the predefined APIs of wpa_supplicant. It is responsible for bi-directional conversion of the MACsec parameters and SONiC database entries.
+
+- The green means these components are in SWSS container. This container uses the SAI APIs to control the MACsec security entities(SecY) according to databases entries and to synchronize the statistics from SecY to COUNTERS_DB.
+  - **MACsecOrch** is a module of orchagent, that uses SAI APIs to manage the SecY according to messages from databases and synchronized the statistics of SecY to COUNTERS_DB.
+
+- The blue one is MACsecSAI in SYNCD container. MACsecSAI is a set of APIs that are defined to communicate with the SecY. In the virtual switch, the SecY is Linux MACsec driver and MACsecSAI will use the ip commands to manage them. But in the real switch, the SecY is the MACsec cipher chip and the implementation of MACsecSAI will be provided by the vendor of the cipher chip.
```

Comment:
>The blue one is MACsecSAI in SYNCD container [](start = 2, length = 44)

The MACsecSAI box in the diagram is confusing. All other boxes are representing processes. And audience keep asking how many syncd processes in your design. You may draw syncd1/sync2 process as boxes, and MACsecSAI/SAI as internal boxes (like the plugin).

Reply:
> > The blue one is MACsecSAI in SYNCD container [](start = 2, length = 44)
> 
> The MACsecSAI box in the diagram is confusing. All other boxes are representing processes. And audience keep asking how many syncd processes in your design. You may draw syncd1/sync2 process as boxes, and MACsecSAI/SAI as internal boxes (like the plugin).

Fixed, please review it.